src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -115,78 +116,76 @@ export function generatePropertyEditorSchema(prop: string, nestedProp: string, p
                                              fieldDef: ShelfFieldDef, shelfId: ShelfId): PropertyEditorSchema {
   const title = generateTitle(prop, nestedProp, propTab);
   const propertyKey = nestedProp ? prop + '_' + nestedProp : prop;
-  const baseUISchema: SchemaProperty = {
-    title: title,
-    type: 'string'
-  };
   const {channel} = shelfId;
   switch (prop) {
     case 'scale':
       const scaleTypes: string[] = getSupportedScaleTypes(shelfId, fieldDef);
-      return generateScaleEditorSchema(nestedProp, scaleTypes, fieldDef, baseUISchema);
+      return generateScaleEditorSchema(nestedProp as keyof Scale, scaleTypes, fieldDef, title, propertyKey);
```

Comment:
@kanitw 

How do I tell Typescript that nestedProp, which is a string, is a `keyof Scale`?